### ISSUE TITLE
jumping to top in rare cases

### ISSUE THREAD

**Issue by axeff** [appbaseio/reactivesearch#528]
**Issue Type:**
bug

**Platform:**
web

**Description:**
When I have several filters I combine, at some point the page jumps to the very top. This not only happens when I receive 0 results but also when I still have results

**Minimal reproduction of the problem with instructions:**
see [sandbox](https://codesandbox.io/s/r53ro5735q). Click on `Nissan` and then `scroll down a little`. Then click on `5` (on the rating component)

**Reactivesearch version:**
2.8.0

**Browser:**
all

**Anything else:**

**Comment by metagrover** [comment]
I can confirm this happens, not sure why though.

Any ideas @divyanshu013?

**Comment by divyanshu013** [comment]
It seems to be because of [this conditional block](https://github.com/appbaseio/reactivesearch/blob/825aec2a028e047dfb3164db614109ebc74c28d7/packages/web/src/components/result/ReactiveList.js#L213-L219):

```js
if (nextProps.hits.length < this.props.hits.length) {
	// query has changed
	window.scrollTo(0, 0);
	this.setState({
		from: 0,
	});
}
```

Not sure about the condition here if its the correct way to determine whether the query has changed. Gist is if next set of hits is lesser than the current ones it will scroll to top.

**Comment by metagrover** [comment]
Ideally, we should wire up that logic to scroll to top if query has changed.

**Comment by divyanshu013** [comment]
I think OP is looking for a way to avoid scrolling to top even if query changes. With `pagination` there is a way to avoid it with `onPageChange`. Perhaps something similar that works with infinite scrolling would be nice.

**Comment by metagrover** [comment]
Maybe we should introduce a prop to handle this and move the scrolling logic out of page change. Because if we use shallow compare, we can handle it from one place, where we will use that boolean prop to determine whether to scroll or not.

**Comment by divyanshu013** [comment]
Sounds good could be a boolean prop to determine whether to scroll to top if query changes. Although not sure if it would be always needed. For example, if page number is changed then you might wanna always scroll to top (or not).

Also this would be a breaking change if we change the `onPageChange` behavior.

**Comment by metagrover** [comment]
Yes, we can keep the prop (`scrollOnChange`) as `true` by default.

**Comment by metagrover** [comment]
Now available in 3.0
